### ISSUE TITLE
New major version required for league/oauth2-server 9

### ISSUE THREAD

**Issue by TomHAnderson** [laravel/passport#1748]
I'm sure other people have reasons to see the upgrade too.  Here's mine:

`league/oauth2-server` 8.x uses `league/event` 2.2 but `league/event` 3.0.x has been out for some time.  My library https://doctrine-orm-graphql.apiskeletons.dev/en/latest/ had to degrade from 3.0.x to 2.2 as a new major version because https://ldog.apiskeletons.dev/ needs an OAuth2 library and since the LDOG Stack is rooted in Laravel, you folks have the best thing going.  

To keep this library as relevant as ever, I'm requesting you begin work for support of `league/oauth2-server` 9

**Comment by Sephster** [comment]
There is an open bug in v9 which I'm hoping to fix early next week. It would be best to get this into passport before a new major is tagged

**Comment by driesvints** [comment]
Hey @TomHAnderson! We have a fully working version ready to go on the `13.x` branch which is pending release. But before we do that there's a couple of things we still want to address. @hafezdivandari is looking into this. A few PR's are still pending a review from Taylor which atm also has quite a few other things going on. I'm also hoping we can release a new major version soon but for now we'll have to have a bit more patience.

@Sephster that's cool. Feel free to ping me here when that's done. Thanks for all your work on oauth server 🙏

**Comment by hafezdivandari** [comment]
@Sephster actually 4 bugs, that I've previously sent PRs to address them:
* thephpleague/oauth2-server#1412
* thephpleague/oauth2-server#1410
* thephpleague/oauth2-server#1449
* thephpleague/oauth2-server#1298

And also 4 more PRs which would be really appreciated if we can merge before major Passport release:
* ~thephpleague/oauth2-server#1446~
* thephpleague/oauth2-server#1420
* thephpleague/oauth2-server#1328
* thephpleague/oauth2-server#1447

**Comment by hafezdivandari** [comment]
Hi @Sephster, I was wondering if you had any availability to review these 2 PRs I've sent?

**Comment by saade** [comment]
Sad to not see Token Introspection in v9 :(

**Comment by driesvints** [comment]
@saade what do you mean?

**Comment by saade** [comment]
@driesvints 

#575 #491

**Comment by driesvints** [comment]
@saade as said in those issues, we need OAuth Server to implement this first before we can implement it in Passport. Maybe that's already done? @Sephster

**Comment by saade** [comment]
Yup, commented on those issues as well, just replied here as @Sephster was aware of this PR :)

**Comment by Sephster** [comment]
Token introspection isn't in v9. No concrete plans on it yet. Working on fixing bugs raised by @hafezdivandari first and will post here when completed. Thanks

**Comment by driesvints** [comment]
Going to close this one now. We'll get a new major version out as soon as we're done with all the needed changes. Thanks
